src/mui/button/SaveButton.js

```diff
@@ -1,26 +1,30 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
+import { submit } from 'redux-form';
 import RaisedButton from 'material-ui/RaisedButton';
 import FlatButton from 'material-ui/FlatButton';
 import ContentSave from 'material-ui/svg-icons/content/save';
 import CircularProgress from 'material-ui/CircularProgress';
 import translate from '../../i18n/translate';
 
-class SaveButton extends Component {
+export class SaveButton extends Component {
 
     handleClick = (e) => {
         if (this.props.saving) {
             // prevent double submission
             e.preventDefault();
+        } else if (!this.props.submitOnEnter) {
+            this.props.dispatch(submit('record-form'));
```

Comment:
It's a good thing to avoid passing dispatch directly in the components. It makes them ignorant about how the submit itself is done